### ISSUE TITLE
`add_startup_stage_before` and `add_startup_stage_after` do not work with core startup stage

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#2229]
## Bevy version

0.5

## Operating system & version

Windows 10

## What you did

use bevy::prelude::*;

```rust
fn main() {
    App::build()
        .add_plugins(DefaultPlugins)
        .add_startup_stage_after(CoreStage::Startup, "new_stage", SystemStage::parallel())
        .run();
}
```

## What you expected to happen

This should compile adding a startup stage after our initial startup stage.

## What actually happened

```
thread 'main' panicked at 'Target stage does not exist: Startup.'
```

## Additional information

This occurs with both before and after, and the same type of error is given for any other core stage. `add_startup_stage` itself works though, and the following variant runs successfully:

```rust
use bevy::prelude::*;

fn main() {
    App::build()
        .add_plugins(DefaultPlugins)
        .add_startup_stage("new_stage", SystemStage::parallel())
        .add_startup_stage_after("new_stage", "newer_stage", SystemStage::parallel())
        .run();
}
```

Changing the order of adding the default plugins vs adding the startup stage doesn't make a difference.

**Comment by Ixentus** [comment]
I've ran into this before. The Startup stages are not under CoreStage, but under StartupStage: https://github.com/bevyengine/bevy/blob/main/crates/bevy_app/src/lib.rs#L51
```rust
.add_startup_stage_after(StartupStage::Startup, "new_stage", SystemStage::parallel())
```

**Comment by alice-i-cecile** [comment]
`CoreStage::Startup` should probably not exist then, and we should document how `add_startup_system_after` should be used :)

**Comment by alice-i-cecile** [comment]
Perhaps we could use another trait to differentiate startup stages from ordinary stages to avoid this sort of error at compile time.

**Comment by bjorn3** [comment]
`CoreStage::Startup` is the stage within which startup stages like the various `StartupStage::*` are added:

https://github.com/bevyengine/bevy/blob/73ae6af6efc3e8e525e8068fa6f58bfd0d03cc8c/crates/bevy_app/src/app_builder.rs#L111-L118

https://github.com/bevyengine/bevy/blob/73ae6af6efc3e8e525e8068fa6f58bfd0d03cc8c/crates/bevy_app/src/app_builder.rs#L231-L246

**Comment by alice-i-cecile** [comment]
Right... I was worried that might be the case. In that case, this is almost certainly just a docs issue in the short term.

**Comment by alice-i-cecile** [comment]
Will be closed by #2434.
